esmvalcore/preprocessor/_bias.py

```diff
@@ -40,79 +66,123 @@ def bias(products, bias_type='absolute', denominator_mask_threshold=1e-3,
         :math:`< 10^{-6}` kg m :math:`^{-2}` s :math:`^{-1}`) it is absolutely
         essential to change the default value in order to get reasonable
         results.
-    keep_reference_dataset: bool, optional (default: False)
+    keep_reference_dataset:
         If ``True``, keep the reference dataset in the output. If ``False``,
-        drop the reference dataset.
+        drop the reference dataset. Ignored if `ref_cube` is given.
 
     Returns
     -------
-    set of esmvalcore.preprocessor.PreprocessorFile
-        Output datasets.
+    set[PreprocessorFile] | CubeList
+        Output datasets/cubes. Will be a :obj:`set` of
+        :class:`~esmvalcore.preprocessor.PreprocessorFile` objects if
+        `products` is also one, a :class:`~iris.cube.CubeList` otherwise.
 
     Raises
     ------
     ValueError
         Not exactly one input datasets contains the facet
-        ``reference_for_bias: true``; ``bias_type`` is not one of
+        ``reference_for_bias: true`` if ``ref_cube=None``; ``ref_cube=None``
+        and the input products are given as iterable of
+        :class:`~iris.cube.Cube` objects; ``bias_type`` is not one of
         ``'absolute'`` or ``'relative'``.
 
     """
-    # Get reference product
-    reference_product = []
-    for product in products:
-        if product.attributes.get('reference_for_bias', False):
-            reference_product.append(product)
-    if len(reference_product) != 1:
-        raise ValueError(
-            f"Expected exactly 1 dataset with 'reference_for_bias: true', "
-            f"found {len(reference_product):d}")
-    reference_product = reference_product[0]
+    ref_product = None
+    all_cubes_given = all(isinstance(p, Cube) for p in products)
 
-    # Extract reference cube
-    # Note: For technical reasons, product objects contain the member
-    # ``cubes``, which is a list of cubes. However, this is expected to be a
-    # list with exactly one element due to the call of concatenate earlier in
-    # the preprocessing chain of ESMValTool. To make sure that this
-    # preprocessor can also be used outside the ESMValTool preprocessing chain,
-    # an additional concatenate call is added here.
-    ref_cube = concatenate(reference_product.cubes)
+    # Get reference cube if not explicitly given
+    if ref_cube is None:
+        if all_cubes_given:
+            raise ValueError(
+                "`ref_cube` cannot be `None` when `products` is an iterable "
+                "of Cubes"
```

Comment:
this is rather cryptic, could you please expand a bit for the user's sake?

Reply:
I tried to improve that in https://github.com/ESMValGroup/ESMValCore/pull/2183/commits/3b8f7d33564bf1965191d74d76512985582f9138

Reply:
biutiful!